src/main.ts

```diff
@@ -3,7 +3,39 @@
 import applyApi from './applyApi'
 import { UserOptions } from './interfaces'
 
+export function applyPlugin(jsPDF) {
+  applyApi(jsPDF)
+}
 export type autoTable = (options: UserOptions) => void
 
-const jsPDF = require('jspdf')
-applyApi(jsPDF)
+let appliedApi = false
+
+try {
+  let jsPDF = require('jspdf')
+  if (jsPDF) {
+    applyApi(jsPDF)
+    appliedApi = true
+  }
+} catch (err) {}
+
+try {
+  let jsPDF = require('jspdf/dist/jspdf.node.min')
+  if (jsPDF) {
+    applyApi(jsPDF)
+    appliedApi = true
+  }
+} catch (err) {}
+
+try {
+  let jsPDF = require('jspdf/dist/jspdf.node.debug')
+  if (jsPDF) {
+    applyApi(jsPDF)
+    appliedApi = true
+  }
+} catch (err) {}
```

Comment:
Here you are trying to load all three `jspdf` dist modules, If the developer is building for the browser and is using a build setup (like webpack) then his bundle will include all three `jspdf` modules (when two of them are for node) adding extra overhead and increasing the bundle size for no reason.

I actually tested it myslef, bundle size increased from `327KB` to `850KB` (minified).

So we should only `require('jspdf')` for the browser version of the plugin, and for the node version we don't `require()` any module and let the developer `applyPlugin()` himself on the `jspdf` module that he imported.

Reply:
💪